src/routes/recovery/recovery.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IAlertsRepository } from '@/domain/alerts/alerts.repository.interface';
+import { AlertsRepository } from '@/domain/alerts/alerts.repository';
+import { AddRecoveryModuleDto } from '@/routes/recovery/entities/add-recovery-module.dto.entity';
+import { Contract } from '@/domain/alerts/entities/alerts.entity';
+
+@Injectable()
+export class RecoveryService {
+  constructor(
+    @Inject(IAlertsRepository)
+    private readonly alertsRepository: AlertsRepository,
+  ) {}
+
+  async addRecoveryModule(args: {
+    chainId: string;
+    safeAddress: string;
+    addRecoveryModuleDto: AddRecoveryModuleDto;
+  }): Promise<void> {
+    const contract: Contract = {
+      chainId: args.chainId,
+      address: args.addRecoveryModuleDto.moduleAddress,
+      displayName: `${args.chainId}:${args.safeAddress}`,
```

Comment:
Is this formatting `<chainId>:<safeAddress>` something that we should enforce across our codebase when using this interface?

Reply:
This is predominantly for Tenderly. We can determine which Safe the module is added to. What are your thoughts?

Whether we include this across the codebase, I am impartial.

Reply:
Out of curiosity: what's the purpose of this `displayName`? I understand this is sent to Tenderly, and what's the purpose there? If it's just for presentation purposes in the UI, I'd concatenate the contract address also, to have an unique identifier. 

And in my opinion, if it's not used in more places in the code, I think the inline template string is clearer.

Reply:
> I understand this is sent to Tenderly, and what's the purpose there? If it's just for presentation purposes in the UI, I'd concatenate the contract address also, to have an unique identifier.

It is just to display it in the UI:

![image](https://github.com/safe-global/safe-client-gateway/assets/20442784/aa0d1868-db53-4d42-ab9f-eec8be675f40)

I moved it back to the payload and added the module address in 890eef9cec76fe8dd7bf39b0bb801e933817beeb.

Reply:
With the "enforcement" that I mentioned it was to consider the following:

```typescript
displayName?: `${string}:${string}:${string}`;
```

instead of

```typescript
displayName?: string;
```

What do you think?

Reply:
It's much better! I've made the types scricter in 1b7d94a3cade2cdf3d86e5c12f982c2099259eec.